platform/src/test/kotlin/org/stellar/anchor/platform/custody/CustodyTransactionServiceTest.kt

```diff
@@ -50,4 +65,85 @@ class CustodyTransactionServiceTest {
     actualCustodyTransaction.createdAt = null
     JSONAssert.assertEquals(entityJson, gson.toJson(entityCapture.captured), JSONCompareMode.STRICT)
   }
+
+  @Test
+  fun test_createPayment_transaction_does_not_exist() {
+    every { custodyTransactionRepo.findById(TRANSACTION_ID) } returns Optional.empty()
+
+    val exception =
+      assertThrows<CustodyNotFoundException> {
+        custodyTransactionService.createPayment(TRANSACTION_ID, REQUEST_BODY)
+      }
+    Assertions.assertEquals("Transaction (id=TRANSACTION_ID) is not found", exception.message)
+    verify(exactly = 0) { paymentService.createTransactionPayment(any(), REQUEST_BODY) }
+  }
+
+  @Test
+  fun test_createPayment_transaction_exists() {
+    val transaction = JdbcCustodyTransaction()
+    every { custodyTransactionRepo.findById(TRANSACTION_ID) } returns Optional.of(transaction)
+    every { custodyTransactionRepo.save(any()) } returns null
+
+    custodyTransactionService.createPayment(TRANSACTION_ID, REQUEST_BODY)
+    verify(exactly = 1) { paymentService.createTransactionPayment(any(), REQUEST_BODY) }
```

Comment:
You can also verify `TRANSACTION_ID` parameter in `createTransactionPayment` method